internal/provider/resources/resource_cluster_workspace_authorization.go

```diff
@@ -0,0 +1,381 @@
+package resources
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/astronomer/terraform-provider-astro/internal/clients"
+	"github.com/astronomer/terraform-provider-astro/internal/clients/platform"
+	"github.com/astronomer/terraform-provider-astro/internal/provider/models"
+	"github.com/astronomer/terraform-provider-astro/internal/provider/schemas"
+	"github.com/astronomer/terraform-provider-astro/internal/utils"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+)
+
+var _ resource.Resource = &ClusterWorkspaceAuthorizationResource{}
+var _ resource.ResourceWithConfigure = &ClusterWorkspaceAuthorizationResource{}
+
+func NewClusterWorkspaceAuthorizationResource() resource.Resource {
+	return &ClusterWorkspaceAuthorizationResource{}
+}
+
+// ClusterWorkspaceAuthorizationResource represents a cluster workspace authorization resource.
+type ClusterWorkspaceAuthorizationResource struct {
+	platformClient *platform.ClientWithResponses
+	organizationId string
+}
+
+func (r *ClusterWorkspaceAuthorizationResource) Metadata(
+	ctx context.Context,
+	req resource.MetadataRequest,
+	resp *resource.MetadataResponse,
+) {
+	resp.TypeName = req.ProviderTypeName + "_cluster_workspace_authorization"
+}
+
+func (r *ClusterWorkspaceAuthorizationResource) Schema(
+	ctx context.Context,
+	req resource.SchemaRequest,
+	resp *resource.SchemaResponse,
+) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "Cluster workspace authorization resource.",
```

Comment:
```suggestion
		MarkdownDescription: "Cluster workspace authorization resource",
```

Reply:
do we usually add periods, if so, ignore comment

Reply:
we don't - changed